src/server/getters/get-markets-info.ts

```diff
@@ -1,23 +1,12 @@
 import { each } from "async";
-import BigNumber from "bignumber.js";
 import * as Knex from "knex";
-import { Address, MarketsRowWithCreationTime, OutcomesRow, UIMarketInfo, UIMarketsInfo, UIOutcomeInfo, ErrorCallback } from "../../types";
+import { Address, MarketsRowWithCreationTime, OutcomesRow, UIMarketsInfo, UIOutcomeInfo, ErrorCallback } from "../../types";
 import { reshapeOutcomesRowToUIOutcomeInfo, reshapeMarketsRowToUIMarketInfo, getMarketsWithReportingState } from "./database";
-import { queryModifier } from "./database";
 
-export function getMarketsInfo(db: Knex, universe: Address|null|undefined, marketIDs: Array<Address>|null|undefined, sortBy: string|null|undefined, isSortDescending: boolean|null|undefined, limit: number|null|undefined, offset: number|null|undefined, callback: (err: Error|null, result?: UIMarketsInfo) => void): void {
-  let query: Knex.QueryBuilder = getMarketsWithReportingState(db).orderBy("creationTime");
-  if (universe == null && marketIDs == null) {
-    return callback(new Error("must include universe or marketIDs parameters"));
-  }
-  if (universe != null) {
-    query = query.where({ universe });
-  }
-  if (marketIDs != null) {
-    query = query.whereIn("markets.marketID", marketIDs);
-  }
-  query = queryModifier(query, "volume", "desc", sortBy, isSortDescending, limit, offset);
-  console.log(query.toSQL());
+export function getMarketsInfo(db: Knex, marketIDs: Array<Address>|null|undefined, sortBy: string|null|undefined, isSortDescending: boolean|null|undefined, limit: number|null|undefined, offset: number|null|undefined, callback: (err: Error|null, result?: UIMarketsInfo) => void): void {
+  let query: Knex.QueryBuilder = getMarketsWithReportingState(db);
+  if (marketIDs == null || !marketIDs.length) return callback(new Error("must include marketIDs parameter"));
+  query = query.whereIn("markets.marketID", marketIDs);
   query.asCallback((err: Error|null, marketsRows?: Array<MarketsRowWithCreationTime>): void => {
```

Comment:
Shouldn't we just also join against the outcomes table instead of doing queries for each market as well?

This is the beginning of an N+1 query problem.

Reply:
The data in outcomes is structured:
```
          outcomes: [{
            id: 0,
            outstandingShares: 1000,
            price: 0.5,
          }, {
            id: 1,
            outstandingShares: 1000,
            price: 0.5,
          }],
```

Reply:
When i've seen this implemented before, it's returning the exact same left-side over and over, and it is up to the client to accumulate. Unless you have a better pattern/toolset to deal with that?

Reply:
It's always up to our client code to structure the data coming back from the DB.

In this case, you can use _.groupBy and _.map to aggregate the outcomes on a market.

The other options is to break this into two queries:

get markets IN list
get outcomes with marketid IN list

and then merge them in your getter logic.